### PR TITLE
Fix `test_dcos_diagnotics` for Windows.

### PR DESCRIPTION
**Pull request by jeschkies**
## High-level description

The Mesos service unit has a different name on Windows so we query that one as well.
The expected files of the diagnostics bundle are different as well.

## Corresponding DC/OS tickets (required)

  - [D2IQ-65707](https://jira.d2iq.com/browse/D2IQ-65707)

### CODE REVIEW

File: packages/dcos-integration-test/extra/test_dcos_diagnostics.py [hunk 1]
```diff
@@ -262,10 +264,13 @@ def get_nodes_from_response(response):
 
         agent_nodes_response = check_json(
             dcos_api_session.health.get('/units/dcos-mesos-slave.service/nodes', node=master))
-
         agent_nodes = get_nodes_from_response(agent_nodes_response)
 
-        assert len(agent_nodes) == len(dcos_api_session.slaves), '{} != {}'.format(agent_nodes, dcos_api_session.slaves)
+        windows_agent_nodes_response = check_json(
+            dcos_api_session.health.get('/units/mesos-agent/nodes', node=master))
```
Comment (jeschkies) [inline packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: @DominikDary and @janisz, this points to inconsistencies between Linux and Windows agents. Ideally the service names would be the same or at least indicate that one is a Linux agent and the other a Windows agent. Do we have a PO for the Windows support?

**Comment by jeschkies** [comment]
It's green!

**Comment by jeschkies** [comment]
@mesosphere-mergebot label Ready for Review

**Comment by jongiddy** [comment]
@mesosphere-mergebot changelog-not-required Change to tests

**Comment by jeschkies** [comment]
@mesosphere-mergebot request-review @janisz

File: packages/dcos-integration-test/extra/test_dcos_diagnostics.py [hunk 2]
```diff
@@ -262,10 +264,17 @@ def get_nodes_from_response(response):
...
-        assert len(agent_nodes) == len(dcos_api_session.slaves), '{} != {}'.format(agent_nodes, dcos_api_session.slaves)
+        # Fetch Windows nodes if we have some.
+        if 'WinRM' in pulled_units:
+            windows_agent_nodes_response = check_json(
+                dcos_api_session.health.get('/units/mesos-agent/nodes', node=master))
+            windows_agent_nodes = get_nodes_from_response(windows_agent_nodes_response)
+        else:
+            windows_agent_nodes = list()
+
+        assert set(agent_nodes + windows_agent_nodes) == set(dcos_api_session.slaves)
```
Comment (DominikDary) [inline packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: can't we add this to test utils?

Comment (janisz) [reply packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: It's not a easy change. We should change the way how we start tests and add new a env variable with windows private nodes. Then we can use it to distinguish IPs in tests.

Comment (jeschkies) [reply packages/dcos-integration-test/extra/test_dcos_diagnostics.py]: Ideally the tests would be mostly agnostic to the node types. Eg the service units should have the same name. I think it is ok to make it explicit right now and not hide it in a library.

**Pull request merged**